fast64_internal/f3d/f3d_gbi.py

```diff
@@ -1984,75 +1984,8 @@ def to_binary(self):
         )
 
     def to_c(self):
-        if bpy.context.scene.decomp_compatible:
-            return (
-                "{{"
-                + ", ".join(
-                    (
-                        (
-                            "{"
-                            + ", ".join(
-                                (
-                                    str(self.position[0]),
-                                    str(self.position[1]),
-                                    str(self.position[2]),
-                                )
-                            )
-                            + "}"
-                        ),
-                        "0",
-                        "{" + str(self.uv[0]) + ", " + str(self.uv[1]) + "}",
-                        (
-                            "{"
-                            + ", ".join(
-                                (
-                                    "0x" + format(self.colorOrNormal[0], "X"),
-                                    "0x" + format(self.colorOrNormal[1], "X"),
-                                    "0x" + format(self.colorOrNormal[2], "X"),
-                                    "0x" + format(self.colorOrNormal[3], "X"),
-                                )
-                            )
-                            + "}"
-                        ),
-                    )
-                )
-                + "}}"
-            )
-        else:
-            return (
-                "{"
-                + ", ".join(
-                    (
-                        str(self.position[0]),
-                        str(self.position[1]),
-                        str(self.position[2]),
-                        "0",
-                        str(self.uv[0]),
-                        str(self.uv[1]),
-                        "0x" + format(self.colorOrNormal[0], "X"),
-                        "0x" + format(self.colorOrNormal[1], "X"),
-                        "0x" + format(self.colorOrNormal[2], "X"),
-                        "0x" + format(self.colorOrNormal[3], "X"),
-                    )
-                )
-                + "}"
-            )
-
-    def to_sm64_decomp_s(self):
-        return "vertex " + ", ".join(
-            (
-                str(self.position[0]),
-                str(self.position[1]),
-                str(self.position[2]),
-                "0",
-                "0x" + format(self.uv[0], "X"),
-                "0x" + format(self.uv[1], "X"),
-                "0x" + format(self.colorOrNormal[0], "X"),
-                "0x" + format(self.colorOrNormal[1], "X"),
-                "0x" + format(self.colorOrNormal[2], "X"),
-                "0x" + format(self.colorOrNormal[3], "X"),
-            )
-        )
+        spc = lambda x: ", ".join([str(a) for a in x])
+        return (f"{{{{ {{{spc(self.position)}}}, 0, {{{spc(self.uv)}}}, {{{spc(self.colorOrNormal)}}} }}}}")
```

Comment:
it's not a dealbreaker to me but I think this is a lot less readable than the previous 30-lines version
I really hate `{` in f-strings
but if others are okay i'm okay

Reply:
I can reduce the number of curly braces by including the braces around parameters inside the unrolling function spc
```
    def to_c(self):
        def spc(x):
            return f"{{', '.join([str(a) for a in x])}}"
        return f"{{{{ {spc(self.position)}, 0, {spc(self.uv)}, {spc(self.colorOrNormal)} }}}}"
```

Reply:
I could also do it this way, which may be the best as it reduces the number of curly braces. Let me know which way is preferred.
```
    def to_c(self):
        def spc(x):
            return "{" + ', '.join([str(a) for a in x] + "}"
        return f"{{{{ {spc(self.position)}, 0, {spc(self.uv)}, {spc(self.colorOrNormal)} }}}}"
        ```

Reply:
The latter one is good, maybe something like `"{{ " + ", ".join([spc(self.position), "0", spc(self.uv), spc(self.colorOrNormal)]) + " }}"`, which is pretty clear about brackets.

Reply:
done